parlai/core/torch_generator_agent.py

```diff
@@ -578,59 +578,36 @@ def set_interactive_mode(self, mode, shared=False):
         else:
             self.skip_generation = self.opt.get('skip_generation', False)
 
-    def _dummy_batch(self, batchsize, maxlen):
+    def _cache_dummy_batch(self, batch: Batch):
         """
-        Create a dummy batch.
-
-        This is used to preinitialize the cuda buffer, or otherwise force a
-        null backward pass after an OOM.
+        Cache a batch to be used as a dummy during _fake_forward_pass.
+        """
+        if not hasattr(self, '_dummy_batch'):
+            self._dummy_batch = batch
```

Comment:
does this unnecessarily use up memory? what if the first batch is huge for example?

Reply:
Could be a worse problem for images. For text, imagine it's a 2x2048x1024 LongTensor (2x for content and label; bs 2048; string length 1024). A long is 8 bytes. So we'll be wasting 32MiB of memory.